### PR TITLE
chore: Refactor `<CalendarListContainer />`

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR refactors the calendar settings page to use the `<CalendarListContainer />` which is used in the installed apps page. This means any changes made to this component should apply to both pages.

This PR also changes the design on the component to match the Figma designs found here: https://www.figma.com/file/xk4HOxtSI82J0F7enMxeak/Live?node-id=7684%3A25506&mode=dev

## Installed Apps Page
CleanShot 2023-11-16 at 12 12 07

## Calendar Settings Page
CleanShot 2023-11-16 at 12 13 23

Fixes # (issue)

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [x] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/web/pages/settings/my-account/calendars.tsx [hunk 1]
```diff
@@ -102,167 +46,9 @@ const CalendarsView = () => {
         CTA={<AddCalendarButton />}
         borderInShellHeader={false}
       />
-      <QueryCell
-        query={query}
-        customLoader={<SkeletonLoader />}
-        success={({ data }) => {
```
Comment (joeauyeung) [inline apps/web/pages/settings/my-account/calendars.tsx:108]: Now using `<CalendarListComponent />`

**Comment by PeerRich** [comment]
good refactor! never understood why we had two different layouts here 🙏

**Comment by ciaranha** [comment]
@joeauyeung looks great. Can you double-check that the top section and bottom are both using `border-subtle`? Looks like bottom might be using `border-default`. `border-default` is reserved for inputs & buttons. Containers should be subtler.

File: apps/web/pages/settings/my-account/calendars.tsx (hunk 1 above)
Comment (sean-brydon) [reply apps/web/pages/settings/my-account/calendars.tsx:108]: 👨‍🍳

**Comment by sean-brydon** [comment]
Looks good! Only issue i've found is when you remove ALL your calendars we dont go back to the empty state and get stuck on this view 
CleanShot 2023-11-17 at 09 16 57@2x

File: apps/web/pages/settings/my-account/calendars.tsx [hunk 2]
```diff
@@ -102,167 +46,9 @@ const CalendarsView = () => {
...
-                  .
-                </Trans>
-              }
-              severity="error"
-            />
-          );
-        }}
-      />
+      <div className="mt-8">
+        <CalendarListContainer />
```
Comment (Udit-takkar) [inline apps/web/pages/settings/my-account/calendars.tsx:50]: NIT: Not blocking

Wrong skeleton

Screenshot 2023-11-17 at 11 24 48 PM

**Review by Udit-takkar — COMMENTED** [review]
@joeauyeung currently the toggle is very slow. we can speed it up by 

1) adding  refetchOnWindowFocus: false in trpc.viewer.connectedCalendars
2) remove await from utils.viewer.connectedCalendars.invalidate() here https://github.com/calcom/cal.com/blob/f2c39fc7861b1f4f32ac07f572d7c9b135b6737e/packages/features/calendars/CalendarSwitch.tsx#L66

Result:-


https://github.com/calcom/cal.com/assets/53316345/b96df1e0-7366-4c15-af80-9b19c870e74c


Before:-

https://github.com/calcom/cal.com/assets/53316345/e357d70c-862d-45b1-bb04-5ee462b36499

**Comment by joeauyeung** [comment]
> @joeauyeung currently the toggle is very slow. we can speed it up by
> 
> 1. adding  refetchOnWindowFocus: false in trpc.viewer.connectedCalendars
> 2. remove await from utils.viewer.connectedCalendars.invalidate() here https://github.com/calcom/cal.com/blob/f2c39fc7861b1f4f32ac07f572d7c9b135b6737e/packages/features/calendars/CalendarSwitch.tsx#L66
> 
> Result:-
> 
>  Screen.Recording.2023-11-17.at.11.37.46.PM.mov 
> Before:-
> 
>  Screen.Recording.2023-11-17.at.11.38.11.PM.mov

Thinking back to this, I think we should keep the invalidate() method here. If something went wrong with toggling the selected calendar, we should accurately show that the calendar is not selected. Since calendar availability plays such an important role I think it's worth sacrificing the time here to ensure the selected calendar is actually selected.

**Review by Udit-takkar — APPROVED** [review]
LGTM

**Pull request merged**